mapie/conformity_scores/conformity_scores.py

```diff
@@ -63,47 +69,62 @@ def get_signed_conformity_scores(
 
         Parameters
         ----------
-        y: NDArray
-            Observed values.
+        X: ArrayLike of shape (n_samples_calib, n_features)
+            Observed feature values.
+
+        y: ArrayLike of shape (n_samples_calib,)
+            Observed target values.
 
-        y_pred: NDArray
-            Predicted values.
+        y_pred: ArrayLike of shape (n_samples_calib,)
+            Predicted target values.
 
         Returns
         -------
-        NDArray
-            Unsigned conformity scores.
+        NDArray of shape (n_samples_calib,)
+            Signed conformity scores.
         """
 
     @abstractmethod
     def get_estimation_distribution(
         self,
+        X: ArrayLike,
         y_pred: ArrayLike,
-        conformity_scores: ArrayLike,
+        conformity_scores: ArrayLike
     ) -> NDArray:
         """
         Placeholder for ``get_estimation_distribution``.
         Subclasses should implement this method!
 
         Compute samples of the estimation distribution from the predicted
-        values and the conformity scores.
+        targets and ``conformity_scores`` that can be either the conformity
+        scores or the quantile of the conformity scores.
 
         Parameters
         ----------
-        y_pred: NDArray
-            Predicted values.
+        X: ArrayLike of shape (n_samples_calib, n_features)
+            Observed feature values.
 
-        conformity_scores: NDArray
-            Conformity scores.
+        y_pred: ArrayLike
+            The shape is either (n_samples_calib, n_samples_train): when the
+            method is called in ``get_bounds`` it needs a prediction per train
+            sample for each test sample to compute the bounds.
+            Or (n_samples_calib, 1): when it is called in ``check_consistency``
+
+        conformity_scores: ArrayLike
+            The shape is either (n_samples_calib, n_alpha) when it is the
+            conformity scores themselves or (n_alpha, 1) when it is only the
+            quantile of the conformity scores.
 
         Returns
         -------
-        NDArray
+        NDArray of shape (n_samples_calib, n_alpha) or
+        (n_samples_calib, n_samples_train) according to the shape of ``y_pred``
             Observed values.
```

Comment:
UPD: prefer to use `n_samples` because there is no notion of calibration in this method (even if in `MapieRegressor` we use it for this purpose).

Reply:
Also, can you add the shape of `y_pred` and `conformity_scores`?

Reply:
In your output, you mention `n_samples_calib` and `n_samples_train`. I suggest to use `n_samples` and `n_references` instead.

Reply:
And you talk about `n_alpha` but it is not mentionned in the inputs. How does it appear? Make a reference in the input doctring for clarity.

Reply:
Do this change also for `check_consistency` method.

Reply:
I don't understand what you mean by n_references

Reply:
You can propose an alternative name if you want. I propose this one because, for each sample, you compute different references.

Reply:
`n_alpha` is mentionned in the description of `conformity_scores`

Reply:
> `n_alpha` is mentionned in the description of `conformity_scores`

You said "The shape is either (n_samples_calib, n_alpha) when it is the conformity scores themselves or (n_alpha, 1) when it is only the quantile of the conformity scores." Why is it not "(1, n_alpha)"?